backend/app/spread/actions.py

```diff
@@ -70,12 +71,22 @@ def update_treasury_assets_with_spread_balances(
                 token_name=spread_token_name,
                 token_symbol=spread_token_symbol,
                 token_address=spread_token_address,
-                balance=balances.balances[spread_token_symbol].loc[end],
+                balance_usd=balances.balances[spread_token_symbol].loc[end],
+                balance=balances.balances[spread_token_symbol].loc[end]
+                / spread_token_usd_quote,
             )
         )
     return treasury
 
 
+def get_usd_quote(
+    spread_token_symbol: str, end: str
```

Comment:
The `get_usd_quote` func is useful only to build the USD value for the whole treasury at end date, so that it appears on the UI in the KPIs components. There is no `start` date consideration here, we only take the quote at `end` date.